media/js/careers/filters.js

```diff
@@ -0,0 +1,152 @@
+(function() {
+    'use strict';
+
+    /**
+     * Monitors inputs and filters a list of positions when they change.
+     */
+    function PositionFilters(typeInput, teamInput, locationInput, positionTable) {
+        this.typeInput = typeInput;
+        this.teamInput = teamInput;
+        this.locationInput = locationInput;
+        this.positionTable = positionTable;
+        this.emptyFilterMessage = document.getElementById('empty-filter-message');
+    }
+
+    PositionFilters.prototype = {
+        /**
+         * Bind onFilterChange to the change events for each input.
+         */
+        bindEvents: function() {
+            var self = this;
+            var callback = function() {
+                self.onFilterChange();
+            };
+
+            this.typeInput.addEventListener('change', callback);
+            this.teamInput.addEventListener('change', callback);
+            this.locationInput.addEventListener('change', callback);
+        },
+
+        /**
+         * When a filter changes, refresh the position list.
+         */
+        onFilterChange: function() {
+            var i;
+            // collection of tr.position elements
+            var positions = this.positionTable.getElementsByClassName('position');
+            var positionsVisible = false;
+            var querystring = '';
+
+            var filters = {
+                'position_type': this.typeInput.value,
+                'team': this.teamInput.value,
+                'location': this.locationInput.value
+            };
+
+            // Hide table and show all positions.
+            this.positionTable.classList.add('hidden');
+            this.emptyFilterMessage.classList.add('hidden');
+
+            for (i = 0; i < positions.length; i++) {
+                positions.item(i).classList.remove('hidden');
+            }
+
+            // Hide positions that don't match the current filters.
+            this.filterPositions('type', filters['position_type']);
+            this.filterPositions('team', filters['team']);
+            this.filterLocations(filters['location']);
+
+            // If there aren't any positions being shown, show the no-results message.
+            for (i = 0; i < positions.length; i++) {
+                if (!positions.item(i).classList.contains('hidden')) {
+                    positionsVisible = true;
+                    break;
+                }
+            }
+
+            if (!positionsVisible) {
+                this.emptyFilterMessage.classList.remove('hidden');
+            }
+
+            // Get rid of unset filters.
+            for (var k in filters) {
+                if (filters.hasOwnProperty(k) && !filters[k]) {
```

Comment:
CircleCi fails on this with
>   73:29  error  Do not access Object.prototype method 'hasOwnProperty' from target object  no-prototype-builtins

```suggestion
                if (Object.prototype.hasOwnProperty.call(filters, k) && !filters[k]) {
```